smac_planner/src/a_star.cpp

```diff
@@ -130,13 +130,13 @@ bool AStarAlgorithm::areInputsValid()
   }
 
   // Check if ending point is valid
-  Node * node;
-  if (getToleranceHeuristic() < 0.001 && !isNodeValid(_goal->getIndex(), node)) {
+  NodePtr node;
+  if (getToleranceHeuristic() < 0.001 && !isNodeValid(_goal->getIndex())) {
```

Comment:
I think the isNodeValid should also be a member of `Node` now. Nice to see that we can remove the `Node *` return type because this new reformat allows us to only access the data structure once as well. (e.g. _goal->isNodeValid()` would be the new syntax.)

Reply:
Initially I intented to do that, the problem was that isNodeValid was introducing circular dependencies between node.hpp and types.hpp. isNodeValid requires the OCCUPIED and INSCRIBED consts defined in types. I could resolve this issue by moving those consts into node.hpp as well.

Reply:
Got it